ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh

```diff
@@ -390,13 +395,35 @@ echo "-------------------"
 cat ${SHARED_DIR}/install-config.yaml | grep -v "password\|username\|pullSecret" | tee ${ARTIFACT_DIR}/install-config.yaml
 
 date "+%F %X" > "${SHARED_DIR}/CLUSTER_INSTALL_START_TIME"
-TF_LOG_PATH="${dir}/terraform.txt"
-export TF_LOG_PATH
+export TF_LOG_PATH="${dir}/terraform.txt"
+
+# Cloud infrastructure problems are common, instead of failing and
+# forcing a retest of the entire job, try the installation again if
+# the installer exits with 4, indicating an infra problem.
+ret=4
+tries=1
+max=3
+set +o errexit
+set +o pipefail
+while [ $ret -eq 4 ] && [ $tries -le $max ]
+do
+  echo "Install attempt $tries of $max"
+  if [ $tries -gt 1 ]; then
+    write_install_status
+    cp "${dir}"/log-bundle-*.tar.gz "${ARTIFACT_DIR}/" 2>/dev/null
+    openshift-install --dir="${dir}" destroy cluster 2>&1 | grep --line-buffered -v 'password\|X-Auth-Token\|UserData:' &
+  else
+    date "+%F %X" > "${SHARED_DIR}/CLUSTER_INSTALL_START_TIME"
+  fi
 
-openshift-install --dir="${dir}" create cluster 2>&1 | grep --line-buffered -v 'password\|X-Auth-Token\|UserData:' &
+  openshift-install --dir="${dir}" create cluster 2>&1 | grep --line-buffered -v 'password\|X-Auth-Token\|UserData:' &
```

Comment:
I feel like we're missing a step here between the destroy and (re)create. When you `destroy`, the assets in the install dir are going to be wiped out. So unless I'm missing something, I suspect when you run the 2nd create the installer is not going to find the install config (or manifests or ignition). 

I think the pattern should be:
1. make copy of install assets
2. run install
3. if failure, destroy cluster
4. retry using copied assets from step 1

There are some subtle pros and cons we could discuss about which install assets (install config, manifests, ignition) are copied.